kafka-connect-ftp/src/test/scala/com/datamountaineer/streamreactor/connect/ftp/source/BySlicesTest.scala

```diff
@@ -169,4 +190,169 @@ class BySlicesTest extends AnyFunSuite with Matchers with BeforeAndAfter with St
+        .updated(FtpSourceConfig.User, "demo")
+        .updated(FtpSourceConfig.Password, "password")
+        .updated(FtpSourceConfig.RefreshRate, "PT0S")
+        .updated(FtpSourceConfig.MonitorTail, "/directory/:sftp_update_slice")
+        .updated(FtpSourceConfig.FileMaxAge, "PT952302H53M5.962S")
+        .updated(FtpSourceConfig.KeyStyle, "struct")
+        .updated(FtpSourceConfig.fileFilter, ".*")
+
+      val cfg = new FtpSourceConfig(configMap.asJava)
+
+      val poller = new FtpSourcePoller(cfg, offsets)
+
+      //push file
+      server.putFile("/directory/file1.txt", fileContent1)
+
+
+      val allReadBytes: Array[Byte] = sftpPollUntilEnd(poller)
+
+      allReadBytes.length shouldBe fileContent1.size
+      allReadBytes.deep shouldBe fileContent1.deep
+
+      logger.info(s"===================================================")
+
+      //append same content to file
+      server.putFile("/directory/file1.txt", fileContent1)
+
+      val allReadBytes1: Array[Byte] = sftpPollUntilEnd(poller)
+
+      //No event is generated
+      allReadBytes1.length shouldBe 0
+    })
+  }
+
+  test("Sftp:Update mode by slices mode with MonitorUpdate and SimpleFileConverter :" +
+    " after update of file, all file data must be sent") {
+    withSftpServer(server => {
+      server.setPort(23)
+      server.addUser("demo", "password")
+
+      val offsets = new DummyOffsetStorage
+      val configMap = Map()
+        .updated(FtpSourceConfig.Address, "localhost:23")
+        .updated(FtpSourceConfig.protocol, "sftp")
+        .updated(FtpSourceConfig.User, "demo")
+        .updated(FtpSourceConfig.Password, "password")
+        .updated(FtpSourceConfig.RefreshRate, "PT0S")
+        .updated(FtpSourceConfig.MonitorUpdate, "/directory/:sftp_update_slice")
+        .updated(FtpSourceConfig.FileMaxAge, "PT952302H53M5.962S")
+        .updated(FtpSourceConfig.KeyStyle, "struct")
+        .updated(FtpSourceConfig.fileFilter, ".*")
+
+      val cfg = new FtpSourceConfig(configMap.asJava)
+
+      val poller = new FtpSourcePoller(cfg, offsets)
+
+      //push file
+      server.putFile("/directory/file1.txt", fileContent1)
+
+
+      val allReadBytes: Array[Byte] = sftpPollUntilEnd(poller)
+
+      allReadBytes.length shouldBe fileContent1.size
+      allReadBytes.deep shouldBe fileContent1.deep
+
+      logger.info(s"===================================================")
+
+      //append content to file
+      val deltaContent = "extra".getBytes
+      server.putFile("/directory/file1.txt", fileContent1 ++ deltaContent)
+
+      val allReadBytes1: Array[Byte] = sftpPollUntilEnd(poller)
+
+      //Only the new delta
+      allReadBytes1.length shouldBe (fileContent1.size + deltaContent.size)
+    })
+  }
+
+  test("Sftp:Update mode by slices mode with MonitorTail and SimpleFileConverter :" +
+    " after update of file, only new data must be sent") {
+    withSftpServer(server => {
+      server.setPort(24)
```

Comment:
Sorry - just realised this.  This is causing me problems running the tests.  Using privileged port numbers will require running the tests as a root user.

From the README of the fake server library that you've used:

> By default the SFTP server listens on an auto-allocated port. During the test this port can be obtained by server.getPort()

I think it would be better to leave them to be auto-allocated?

Reply:
Done, and uploaded. There's only one test that set the port to 22, and is mandatory to prove that when you don't specify a port in address, is the 22 the one used.